contrib/RL_out_sharp.lua

```diff
@@ -0,0 +1,208 @@
+    * 20200308 - initial version
+  
+]]
+
+local dt = require "darktable"
+local du = require "lib/dtutils"
+local df = require "lib/dtutils.file"
+local dtsys = require "lib/dtutils.system"
+
+-- module name
+local MODULE_NAME = "RL_out_sharp"
+
+-- check API version
+du.check_min_api_version("5.0.2", MODULE_NAME)  -- darktable 3.x
+
+-- OS compatibility
+local PS = dt.configuration.running_os == "windows" and  "\\"  or  "/"
+
+-- translation
+local gettext = dt.gettext
+gettext.bindtextdomain(MODULE_NAME, dt.configuration.config_dir..PS.."lua"..PS.."locale"..PS)
+local function _(msgid)
+  return gettext.dgettext(MODULE_NAME, msgid)
+  end
+
+-- initialize module preferences
+if not dt.preferences.read(MODULE_NAME, "initialized", "bool") then
+  dt.preferences.write(MODULE_NAME, "sigma", "string", "0.7")
+  dt.preferences.write(MODULE_NAME, "iterations", "string", "10")
+  dt.preferences.write(MODULE_NAME, "jpg_quality", "string", "95")
+  dt.preferences.write(MODULE_NAME, "initialized", "bool", true)
+  end 
+
+
+-- setup export ---------------------------------------------------------------
+local function setup_export(storage, img_format, image_table, high_quality, extra_data)
+  -- set 16bpp if format is tif
+  if img_format.extension == "tif" then
+    img_format.bpp = 16
+    end
+  end
+
+
+-- temp export formats: jpg and tif are supported -----------------------------
+local function supported(storage, img_format)
+  return (img_format.extension == "jpg") or (img_format.extension == "tif")
+  end
+
+
+-- export and sharpen images --------------------------------------------------
+local function export2RL(storage, image_table, extra_data) 
+
+  local temp_name, new_name, run_cmd, result
+  local input_file, output_file, options
+
+  -- read parameters
+  local gmic = df.sanitize_filename(dt.preferences.read(MODULE_NAME, "gmic_exe", "string"))
+  local output_folder = output_folder_selector.value
+  local sigma_str = string.gsub(string.format("%.2f", sigma_slider.value), ",", ".")
+  local iterations_str = string.format("%.0f", iterations_slider.value)
+  local jpg_quality_str = string.format("%.0f", jpg_quality_slider.value)
+
+  -- save preferences
+  dt.preferences.write(MODULE_NAME, "sigma", "string", sigma_str)
+  dt.preferences.write(MODULE_NAME, "iterations", "string", iterations_str)
+  dt.preferences.write(MODULE_NAME, "jpg_quality", "string", jpg_quality_str)
+
+  local gmic_operation = " -deblur_richardsonlucy "..sigma_str..","..iterations_str..",1"
+ 
+  local i = 0
+  for image, temp_name in pairs(image_table) do
+
+    i = i + 1
+    dt.print(_("sharpening image ")..i.." ...")
+    -- create unique filename
+    new_name = output_folder..PS..df.get_basename(temp_name)..".jpg"
+    while df.check_if_file_exists(new_name) do
+      new_name = df.filename_increment(new_name)
+      -- limit to 50 more exports of the original export
+      if string.match(df.get_basename(new_name), "_%d%d$") == "_50" then break end
+      end
```

Comment:
There is a function df.create_unique_filename() that does this but it goes to 99.

Reply:
Yes, this comes from my other script and when I tried to use df.create_unique_filename() it didn't work for some reason, so I implemented myself.
I will try again

Reply:
If there is a problem with create_unique_filename, let me know and I'll track it down.